lib/beacon/css_compiler.ex

```diff
@@ -9,29 +9,37 @@ defmodule Beacon.CSSCompiler do
 
   @behaviour Beacon.RuntimeCSS
 
+  @template_tailwind_config_path Path.join("assets", "tailwind.config.js.eex")
+  @input_css_path Path.join(["assets", "css", "app.css"])
+
+  @external_resource @template_tailwind_config_path
+  @external_resource @input_css_path
+
   @impl Beacon.RuntimeCSS
   def compile!(%Layout{} = layout, opts \\ []) do
-    input_file = Path.join(:code.priv_dir(:beacon), "assets/beacon.css")
+    Application.put_env(:tailwind, :version, Beacon.tailwind_version())
```

Comment:
I'm not a fan of libraries modifying app env vars on the fly. As a developer I would be upset if my app env vars changed without me changing them. Setting it if it isn't set already is still less than idea, but better than stomping on them. How about this?

```suggestion
    required_tailwind = Beacon.tailwind_version()
    case Application.get_env(:tailwind, :version, nil) do
      nil -> Application.put_env(:tailwind, :version, required_tailwind)
      ^required_version -> nil
      other -> raise "Beacon requires Tailwind version #{required_tailwind} but found #{other}"
    end
```

Reply:
Yeah definitely it shouldn't change app's config silently :)

I've changed to just check if it's present instead of forcing a specific version.